src/system/file_path.hh

```diff
@@ -19,6 +19,7 @@
 #define FILE_NAME_HH_
 
 #include <string>
+#include <boost/filesystem.hpp>
```

Comment:
This include is really huge. To avoid it:
1. make output_path etc. shared_ptr<boost::fielsystem::path> and use make_shared for initialization 
2. use just forward declaration of boost::filesystem::path i.e.
	namespace boost {
		namespace filesystem  {
			class path
	}}